packages/react/src/types.ts

```diff
@@ -9,3 +9,5 @@ export interface ProgressivelyProviderProps {
   websocketUrl?: string;
   children?: React.ReactNode;
 }
+
+export type StateMachineConstants = "idle" | "loading" | "success" | "failure";
```

Comment:
is that helpful to have both success and failure when failure can be determined checking the truthy / falsy value of the 'error' object?

Reply:
I would say yes 😄 
This contant represents the current status of your request, and **failure** is one of them.

It's about "what should be in state and what can be computed from state".
If we think that an `isError` can be the result of a `!error`, with the same logic we could define a `isLoading` that could be the result of a `!error && !flags`.
If you want to add another state, you'll need to update all variables that are combinations of state values.
Whereas, if you want to add a possible state the the enum, it's not a problem because it will never enter in conflict with other values.

Don't know if I'm clear but here two links that brainwashed myself 😅 
- https://beta.reactjs.org/learn/reacting-to-input-with-state
- https://redux.js.org/style-guide/#treat-reducers-as-state-machines